optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -354,7 +353,7 @@ def pre_attn_forward(
 
         kv_seq_len = key_states.shape[-2]
         if past_key_value is not None:
-            if token_idx is None:
+            if token_idx is None or past_key_value[0].shape[-2] < token_idx:
                 if hasattr(past_key_value, "get_usable_length"):
```

Comment:
Why do we need this change? past_key_value[0].shape[-2] < token_idx:

Reply:
thanks for the remind, I update the PR to support reuse_cache, bucket_size optimization in prompt/ptuning/prefix ptuning generation. past_key_value[0].shape[-2] < token_idx. this is the case for prefix tuning peft text-generation. which extra pask_key_value will be attached to generate the 1st token.